web-client/src/components/AddSockForm.tsx

```diff
@@ -0,0 +1,266 @@
+    onClose?: () => void;
+}
+
+interface FormData {
+    name: string;
+    description: string;
+    sizes: {
+        size: SockSize;
+        quantity: number;
+        price: number;
+    }[];
+    previewImageUrl: string;
+}
+
+export default function AddSockForm({ onAddSock, onClose }: AddSockFormProps) {
+    const [step, setStep] = useState(1);
+
+
+    const { mutate: createSock } = useCreateSockMutation();
+
+    const {
+        register,
+        control,
+        handleSubmit,
+        watch,
+        formState: { errors, isValid },
+    } = useForm<FormData>({
+        mode: "onChange",
+        defaultValues: {
+            name: "",
+            description: "",
+            sizes: [],
+            previewImageUrl: "",
+        },
+    });
+
+    const { fields, append, remove } = useFieldArray({
+        control,
+        name: "sizes",
+    });
+
+    const selectedSizes = watch("sizes").map((item) => item.size);
+
+    const onSubmit = (data: FormData) => {
+
+        const payload = {
+            sock: {
+                name: data.name.trim(),
+                description: data.description.trim(),
+                previewImageUrl: data.previewImageUrl.trim(),
+            },
+            variants: data.sizes.map((variant) => ({
+                size: variant.size,
+                price: Number(variant.price),
+                quantity: Number(variant.quantity),
+            })),
+        };
+
+        console.log("Payload being sent to API:", payload);
+
+        createSock(payload, {
+            onSuccess: (response) => {
+                console.log("Sock created successfully:", response);
+                onAddSock(response);
+                onClose?.();
+            },
+            onError: (error: any) => {
+                if (error.response?.data) {
+                    console.error("Validation error from server:", error.response.data);
+                } else {
+                    console.error("Unexpected error:", error.message);
+                }
+            },
+        });
+    };
+
+    const nextStep = () => setStep((prev) => prev + 1);
+    const prevStep = () => setStep((prev) => prev - 1);
+
+    return (
+        <form
+            onSubmit={handleSubmit(onSubmit)}
+            className="space-y-6 p-4 bg-white border rounded-md max-w-lg mx-auto text-black"
+        >
+            <h2 className="text-xl font-bold text-center">
+                Add New Item - Step {step} of 3
+            </h2>
+
+
+            {step === 1 && (
+                <div className="space-y-4">
+                    <Input
+                        {...register("name", { required: "Name is required" })}
+                        placeholder="Name"
+                        className="w-full border-black"
+                    />
+                    {errors.name && (
+                        <p className="text-red-500 text-sm">{errors.name.message}</p>
+                    )}
+
+                    <Input
```

Comment:
This should really be a `textarea` so you can just update to be `<Textarea ...`

This way is easier to input longer pieces of text.